### ISSUE TITLE
installing on Macbook with M1 chip

### ISSUE THREAD

**Issue by code86** [ish-app/ish#1382]
I am using a Macbook (with M1 chip), and cannot download this app from Apple App Store (message is "This app is available only on the App Store for iPhone and iPad.").
 
I have been using various iOS apps on Macbook M1, and really hope to do so with **ish** too.

Is this due to a technical limitation, or is it some settings that prevent  it from being downloaded to MacBook M1?

**Comment by tbodt** [comment]
iSH isn't available on Mac because frankly it's almost useless when Terminal exists. If you still want to use it, you can build from source.

**Comment by code86** [comment]
actually, I find iSH to be remarkably useful, as a "sandbox" environment for me to learn Linux. It's very easy + fast to install on iPhone, and works remarkably well. 

Do you have suggestions of any sandbox environment that allows me to learn/play with Linux? I tried UTM + ArchLinux on my Apple Silicon, but their default image seems outdated, and I have to update (`pacman -Syu`) before I can even install tools like GNU Emacs (`pacman -S emacs`). As a comparison, iSH installs Emacs directly without any errors with the `apk add` command! Great job, iSH!

If there is no technical limitation, I think allowing iSH to be installed on Apple Silicon (from App Store) would be a great help for novices like me to begin exploring GNU/Linux. Thanks for the wonderful iSH app!

**Comment by tbodt** [comment]
I think UTM is a good option. It should be possible to use Alpine too, if you download a "virtual" image from alpinelinux.org.

**Comment by Nio-av** [comment]
> If there is no technical limitation, I think allowing iSH to be installed on Apple Silicon (from App Store) would be a great help for novices like me to begin exploring GNU/Linux.

I totally agree. Especially for public courses or in schools this could be a really interesting option. Setting up a virtual machine; downloading linux & installing the OS is a time-consuming way. Just downloading one small app could be an attractive alternative.

**Comment by krumble1** [comment]
I'm also curious about this, purely from an experimental, boundary-pushing viewpoint. @tbodt if some technical problem or other issue is preventing installation on an M1 Mac, would you mind briefly sharing a bit about those difficulties?

**Comment by jankais3r** [comment]
> iSH isn't available on Mac because frankly it's almost useless when Terminal exists. If you still want to use it, you can build from source.

I would love to have iSH on macOS. Sometimes you don't want to "litter" your main OS with random stuff and packages you are currently experimenting with. And while [VM](https://gist.github.com/jankais3r/238da941ba6a6455c5f81a07cc2cfd18) is an option, the instant "boot" and small RAM footprint of iSH would make it great. Like WSLv1 on Windows.

**Comment by totoLab** [comment]
> iSH isn't available on Mac because frankly it's almost useless when Terminal exists. If you still want to use it, you can build from source.

Currently, I'm taking a microprogramming course at my university and my M1 mac's owners' colleagues are struggling with emulating x86 Linux VM (the professor is going to help only if you use Linux x86 instruction set), so an app would be very useful for them.

**Comment by IanEdington** [comment]
@totoLab awesome! Best of luck with your course.

Learning docker (or another container platform) would serve these purposes a lot better and would be more useful in your professional careers. Not to mention having a LOT more resources online when you get stumped by something. Containers also has an almost instant spin up time once the initial build is finished.

> ditch the Docker for Mac virtual machine (with its IO perf issues)

I guarantee you’ll have more issues and fewer resources with a less mature tool.

Don’t hear me wrong, iSH is awesome. Just not for this.

**Comment by jankais3r** [comment]
> Currently, I'm taking a microprogramming course at my university and my M1 mac's owners' colleagues are struggling with emulating x86 Linux VM (the professor is going to help only if you use Linux x86 instruction set), so an app would be very useful for them.

Tell your colleagues to take a look at [UTM](https://github.com/utmapp/UTM/releases/latest). It's a nice GUI for the QEMU backend, which means that it can emulate many architectures incl. x86. iSH is great for running tools that you know work elsewhere, but since it can't (yet) emulate the complete x86 instruction set, it wouldn't be my first choice for active development work. Especially on macOS where there are great alternatives. Also, iSH runs Alpine which is muslc-based, which might again cause unexpected results during your exercises.

**Comment by tbodt** [comment]
Since it sounds like it would actually be useful for a few things, I went ahead and enabled it in the app store. I wouldn't expect it to be a particularly great experience though, since the UI is only tested on iOS and not macOS.

**Comment by jankais3r** [comment]
Thank you @tbodt. The UI works well, but selecting a text with mouse and then right-clicking crashes the app.

Let me know if you want me to open a separate Issue for it. Right now I am assuming that you might not want to spend time on supporting the macOS build, so I'll just paste it here.

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Incident Identifier: 245E6BA6-B1E5-4AB1-897B-0320AF26686F
CrashReporter Key:   9778F5AA-205D-100A-1057-C2B4B768E788
Hardware Model:      MacBookPro18,4
Process:             iSH [77790]
Path:                /private/var/folders/*/iSH.app/iSH
Identifier:          app.ish.iSH
Version:             1.2.3 (298)
AppStoreTools:       13C90b
AppVariant:          1:MacFamily20,1:15
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           app.ish.iSH [35967]

Date/Time:           2022-05-14 14:26:56.9839 +0200
Launch Time:         2022-05-14 14:18:54.7019 +0200
OS Version:          macOS 12.3.1 (21E258)
Release Type:        User
Report Version:      104

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000001, 0x00000001b2769514
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: SIGNAL 5 Trace/BPT trap: 5
Terminating Process: exc handler [77790]

Triggered by Thread:  0

Application Specific Information:
Crashing on exception: *** -[__NSDictionaryM setObject:forKey:]: object cannot be nil (key: title)


Last Exception Backtrace:
0   CoreFoundation                	       0x1af90eaf4 __exceptionPreprocess + 220
1   libobjc.A.dylib               	       0x1af659e14 objc_exception_throw + 60
2   CoreFoundation                	       0x1af9e0384 -[__NSCFString characterAtIndex:].cold.1 + 0
3   CoreFoundation                	       0x1af9ed198 -[__NSDictionaryM setObject:forKey:].cold.3 + 0
4   CoreFoundation                	       0x1af847164 -[__NSDictionaryM setObject:forKey:] + 904
5   UIKitCore                     	       0x1d791c510 -[_UIMenuBarItem properties] + 124
6   UIKitMacHelper                	       0x1c5111898 UINSNSMenuItemFromUINSMenuItem + 96
7   UIKitMacHelper                	       0x1c511159c _insertUINSMenuItemsIntoNSMenu + 852
8   UIKitMacHelper                	       0x1c5110ff4 UINSNSMenuFromUINSMenu + 152
9   UIKitMacHelper                	       0x1c5110ec4 -[UINSMenuController _createNSMenu:forContextMenu:] + 92
10  UIKitMacHelper                	       0x1c51831d4 -[UINSMenuController _prepareToShowContextMenu:activityItemsConfiguration:] + 144
11  UIKitMacHelper                	       0x1c5182b34 -[UINSMenuController showContextMenu:inWindow:atLocationInWindow:activityItemsConfiguration:] + 312
12  libdispatch.dylib             	       0x1af5fc604 _dispatch_call_block_and_release + 32
13  libdispatch.dylib             	       0x1af5fe1c8 _dispatch_client_callout + 20
14  libdispatch.dylib             	       0x1af60c6cc _dispatch_main_queue_drain + 928
15  libdispatch.dylib             	       0x1af60c31c _dispatch_main_queue_callback_4CF + 44
16  CoreFoundation                	       0x1af8d03e4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
17  CoreFoundation                	       0x1af88d614 __CFRunLoopRun + 2532
18  CoreFoundation                	       0x1af88c490 CFRunLoopRunSpecific + 600
19  HIToolbox                     	       0x1b84acdd8 RunCurrentEventLoopInMode + 292
20  HIToolbox                     	       0x1b84acb54 ReceiveNextEventCommon + 564
21  HIToolbox                     	       0x1b84ac908 _BlockUntilNextEventMatchingListInModeWithFilter + 72
22  AppKit                        	       0x1b23ebb2c _DPSNextEvent + 860
23  AppKit                        	       0x1b23ea424 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1328
24  AppKit                        	       0x1b23dc5f0 -[NSApplication run] + 596
25  AppKit                        	       0x1b23add08 NSApplicationMain + 1132
26  AppKit                        	       0x1b2683dd0 +[NSWindow _savedFrameFromString:] + 0
27  UIKitMacHelper                	       0x1c510ea7c UINSApplicationMain + 1276
28  UIKitCore                     	       0x1d77a99f8 UIApplicationMain + 164
29  iSH                           	       0x102c428a8 0x102c38000 + 43176
30  dyld                          	       0x102fe5088 start + 516

Thread 0 Crashed:: login-7 Dispatch queue: com.apple.main-thread
0   AppKit                        	       0x1b2769514 -[NSApplication _crashOnException:] + 324
1   AppKit                        	       0x1b276950c -[NSApplication _crashOnException:] + 316
2   AppKit                        	       0x1b276922c -[NSApplication reportException:] + 580
3   AppKit                        	       0x1b2813708 uncaughtErrorProc + 156
4   CoreFoundation                	       0x1af99abac __handleUncaughtException + 676
5   libobjc.A.dylib               	       0x1af662324 _objc_terminate() + 132
6   libc++abi.dylib               	       0x1af77bea4 std::__terminate(void (*)()) + 20
7   libc++abi.dylib               	       0x1af77be40 std::terminate() + 64
8   libdispatch.dylib             	       0x1af5fe1dc _dispatch_client_callout + 40
9   libdispatch.dylib             	       0x1af60c6cc _dispatch_main_queue_drain + 928
10  libdispatch.dylib             	       0x1af60c31c _dispatch_main_queue_callback_4CF + 44
11  CoreFoundation                	       0x1af8d03e4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
12  CoreFoundation                	       0x1af88d614 __CFRunLoopRun + 2532
13  CoreFoundation                	       0x1af88c490 CFRunLoopRunSpecific + 600
14  HIToolbox                     	       0x1b84acdd8 RunCurrentEventLoopInMode + 292
15  HIToolbox                     	       0x1b84acb54 ReceiveNextEventCommon + 564
16  HIToolbox                     	       0x1b84ac908 _BlockUntilNextEventMatchingListInModeWithFilter + 72
17  AppKit                        	       0x1b23ebb2c _DPSNextEvent + 860
18  AppKit                        	       0x1b23ea424 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1328
19  AppKit                        	       0x1b23dc5f0 -[NSApplication run] + 596
20  AppKit                        	       0x1b23add08 NSApplicationMain + 1132
21  AppKit                        	       0x1b2683dd0 _NSApplicationMainWithInfoDictionary + 24
22  UIKitMacHelper                	       0x1c510ea7c UINSApplicationMain + 1276
23  UIKitCore                     	       0x1d77a99f8 UIApplicationMain + 164
24  iSH                           	       0x102c428a8 0x102c38000 + 43176
25  dyld                          	       0x102fe5088 start + 516

Thread 1:: com.apple.uikit.eventfetch-thread
0   libsystem_kernel.dylib        	       0x1af7848d0 mach_msg_trap + 8
1   libsystem_kernel.dylib        	       0x1af784d40 mach_msg + 76
2   CoreFoundation                	       0x1af88ec1c __CFRunLoopServiceMachPort + 372
3   CoreFoundation                	       0x1af88d0cc __CFRunLoopRun + 1180
4   CoreFoundation                	       0x1af88c490 CFRunLoopRunSpecific + 600
5   Foundation                    	       0x1b0769210 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 236
6   Foundation                    	       0x1b07f425c -[NSRunLoop(NSRunLoop) runUntilDate:] + 124
7   UIKitCore                     	       0x1d77ab174 -[UIEventFetcher threadMain] + 156
8   Foundation                    	       0x1b076171c __NSThread__start__ + 808
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 2:: init-1
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c636c0 0x102c38000 + 177856
5   iSH                           	       0x102c5ac30 0x102c38000 + 142384
6   iSH                           	       0x102c5baa8 0x102c38000 + 146088
7   iSH                           	       0x102c5bb90 0x102c38000 + 146320
8   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
9   libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 3:: getty-5
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c73c3c 0x102c38000 + 244796
5   iSH                           	       0x102c66078 0x102c38000 + 188536
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 4:: getty-6
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c73c3c 0x102c38000 + 244796
5   iSH                           	       0x102c66078 0x102c38000 + 188536
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 5:: com.apple.NSEventThread
0   libsystem_kernel.dylib        	       0x1af7848d0 mach_msg_trap + 8
1   libsystem_kernel.dylib        	       0x1af784d40 mach_msg + 76
2   CoreFoundation                	       0x1af88ec1c __CFRunLoopServiceMachPort + 372
3   CoreFoundation                	       0x1af88d0cc __CFRunLoopRun + 1180
4   CoreFoundation                	       0x1af88c490 CFRunLoopRunSpecific + 600
5   AppKit                        	       0x1b25586f4 _NSEventThread + 196
6   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
7   libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 6:: login-7
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c5e9f8 0x102c38000 + 158200
5   iSH                           	       0x102c5eda4 0x102c38000 + 159140
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 7:: ash-8
0   libsystem_kernel.dylib        	       0x1af78ac40 kevent + 8
1   iSH                           	       0x102c76f54 0x102c38000 + 257876
2   iSH                           	       0x102c77a2c 0x102c38000 + 260652
3   iSH                           	       0x102c5ac30 0x102c38000 + 142384
4   iSH                           	       0x102c5baa8 0x102c38000 + 146088
5   iSH                           	       0x102c5bb90 0x102c38000 + 146320
6   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
7   libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 8:: getty-11
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c73c3c 0x102c38000 + 244796
5   iSH                           	       0x102c66078 0x102c38000 + 188536
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 9:: getty-12
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c73c3c 0x102c38000 + 244796
5   iSH                           	       0x102c66078 0x102c38000 + 188536
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 10:: getty-13
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c73c3c 0x102c38000 + 244796
5   iSH                           	       0x102c66078 0x102c38000 + 188536
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 11:: getty-14
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   iSH                           	       0x102c78610 0x102c38000 + 263696
3   iSH                           	       0x102c784f4 0x102c38000 + 263412
4   iSH                           	       0x102c73c3c 0x102c38000 + 244796
5   iSH                           	       0x102c66078 0x102c38000 + 188536
6   iSH                           	       0x102c5ac30 0x102c38000 + 142384
7   iSH                           	       0x102c5baa8 0x102c38000 + 146088
8   iSH                           	       0x102c5bb90 0x102c38000 + 146320
9   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
10  libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 12:
0   libsystem_pthread.dylib       	       0x1af7bd078 start_wqthread + 0

Thread 13:
0   libsystem_pthread.dylib       	       0x1af7bd078 start_wqthread + 0

Thread 14:: JavaScriptCore libpas scavenger
0   libsystem_kernel.dylib        	       0x1af788290 __psynch_cvwait + 8
1   libsystem_pthread.dylib       	       0x1af7c283c _pthread_cond_wait + 1236
2   JavaScriptCore                	       0x1e849634c scavenger_thread_main + 1220
3   libsystem_pthread.dylib       	       0x1af7c226c _pthread_start + 148
4   libsystem_pthread.dylib       	       0x1af7bd08c thread_start + 8

Thread 15:
0   libsystem_pthread.dylib       	       0x1af7bd078 start_wqthread + 0

Thread 16:
0   libsystem_pthread.dylib       	       0x1af7bd078 start_wqthread + 0

Thread 17:
0   libsystem_pthread.dylib       	       0x1af7bd078 start_wqthread + 0

Thread 18:
0   libsystem_pthread.dylib       	       0x1af7bd078 start_wqthread + 0


Thread 0 crashed with ARM Thread State (64-bit):
    x0: 0x0000000149915c00   x1: 0x000000016d1c5140   x2: 0x0000000000000004   x3: 0x0000000149916600
    x4: 0x00000000000001d3   x5: 0x000000016d1c5088   x6: 0x0000000000000005   x7: 0x00000001afa7b7dc
    x8: 0x0000000205811000   x9: 0xcd606f9e0fc400e8  x10: 0x0000000000000001  x11: 0x00000001499167d3
   x12: 0x00000000fca02400  x13: 0x0000000000000002  x14: 0x000000000000fffd  x15: 0x0000000000000003
   x16: 0x0000000000000001  x17: 0x000000016d1c54e6  x18: 0x0000000000000000  x19: 0x0000000141b88dc0
   x20: 0x00000001f9887cce  x21: 0x0000000147a15510  x22: 0x00000001f9882000  x23: 0x0000000207f4e000
   x24: 0x0000000141b7cb00  x25: 0x0000000141b1e000  x26: 0x00000001f9883399  x27: 0x00000001f9884d11
   x28: 0x00000001afcac80c   fp: 0x000000016d1c5b40   lr: 0x7a150001b276950c
    sp: 0x000000016d1c5af0   pc: 0x00000001b2769514 cpsr: 0x60001000
   far: 0x0000000149a47208  esr: 0xf2000001 (Breakpoint) brk 1

Binary Images:
       0x1b23aa000 -        0x1b3261fff com.apple.AppKit (6.9) <153590c1-624c-38fb-8ffb-ba751bccbd16> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
       0x1af80a000 -        0x1afd4ffff com.apple.CoreFoundation (6.9) <8e1809ff-f5b1-3d9e-be15-f2e7b1d8ce51> /System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation
       0x1af641000 -        0x1af67efff libobjc.A.dylib (*) <4b5ab8fb-60e0-3992-ba0e-a073a3fa86a7> /usr/lib/libobjc.A.dylib
       0x1af76b000 -        0x1af782fff libc++abi.dylib (*) <e0fc64a3-790d-34b3-a092-432a0e6b1c82> /usr/lib/libc++abi.dylib
       0x1af5fa000 -        0x1af640fff libdispatch.dylib (*) <dc048e3b-e023-3d17-afe5-4ff3dc625608> /usr/lib/system/libdispatch.dylib
       0x1b8479000 -        0x1b87acfff com.apple.HIToolbox (2.1.1) <4a1d1f30-c79d-3502-b51f-9792adbd8e06> /System/Library/Frameworks/Carbon.framework/Versions/A/Frameworks/HIToolbox.framework/Versions/A/HIToolbox
       0x1c510a000 -        0x1c520ffff com.apple.UIKitMacHelper (1.0) <cf1d8545-c12b-3515-9519-b9840e38052c> /System/Library/PrivateFrameworks/UIKitMacHelper.framework/Versions/A/UIKitMacHelper
       0x1d77a6000 -        0x1d8eb0fff com.apple.UIKitCore (1.0) <aa248023-fcd7-308f-bcdc-b93116ba5a2b> /System/iOSSupport/System/Library/PrivateFrameworks/UIKitCore.framework/Versions/A/UIKitCore
       0x102c38000 -        0x102cdbfff app.ish.iSH (1.2.3) <5f72cbc8-f1ca-34a3-8882-bf0a879afe83> /private/var/folders/*/iSH.app/iSH
       0x102fe0000 -        0x10303ffff dyld (*) <fbb89662-e6f2-3434-b542-f75185ac5e74> /usr/lib/dyld
       0x1af783000 -        0x1af7bafff libsystem_kernel.dylib (*) <1d7b3b8e-75a1-34ea-aa52-9f7c23155c55> /usr/lib/system/libsystem_kernel.dylib
       0x1b0705000 -        0x1b0af5fff com.apple.Foundation (6.9) <886a3487-88a6-30bb-b179-1a92996c67cf> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
       0x1af7bb000 -        0x1af7c7fff libsystem_pthread.dylib (*) <cee8bc77-6923-34d9-89a3-6f8f7279605e> /usr/lib/system/libsystem_pthread.dylib
       0x1e7245000 -        0x1e85c0fff com.apple.JavaScriptCore (17613) <f13513f2-9b55-396d-b981-add74643fb96> /System/iOSSupport/System/Library/Frameworks/JavaScriptCore.framework/Versions/A/JavaScriptCore

EOF

-----------
Full Report
-----------

{"app_name":"iSH","timestamp":"2022-05-14 14:26:59.00 +0200","app_version":"1.2.3","slice_uuid":"5f72cbc8-f1ca-34a3-8882-bf0a879afe83","adam_id":"1436902243","build_version":"298","platform":2,"bundleID":"app.ish.iSH","share_with_app_devs":0,"is_first_party":0,"bug_type":"309","os_version":"macOS 12.3.1 (21E258)","incident_id":"245E6BA6-B1E5-4AB1-897B-0320AF26686F","name":"iSH"}
{
  "uptime" : 420000,
  "procLaunch" : "2022-05-14 14:18:54.7019 +0200",
  "procRole" : "Foreground",
  "version" : 2,
  "userID" : 501,
  "deployVersion" : 210,
  "modelCode" : "MacBookPro18,4",
  "procStartAbsTime" : 10137625994332,
  "coalitionID" : 35967,
  "osVersion" : {
    "train" : "macOS 12.3.1",
    "build" : "21E258",
    "releaseType" : "User"
  },
  "captureTime" : "2022-05-14 14:26:56.9839 +0200",
  "incident" : "245E6BA6-B1E5-4AB1-897B-0320AF26686F",
  "bug_type" : "309",
  "pid" : 77790,
  "procExitAbsTime" : 10149199071283,
  "translated" : false,
  "cpuType" : "ARM-64",
  "procName" : "iSH",
  "procPath" : "\/private\/var\/folders\/*\/iSH.app\/iSH",
  "bundleInfo" : {"CFBundleShortVersionString":"1.2.3","CFBundleVersion":"298","CFBundleIdentifier":"app.ish.iSH","DTAppStoreToolsBuild":"13C90b"},
  "storeInfo" : {"storeCohortMetadata":"10|date=1652529600000&sf=143441&pgtp=Software&pgid=1436902243&prpg=Search_ios&ctxt=Search&issrch=1&lngid=1","itemID":"1436902243","deviceIdentifierForVendor":"60C89302-4C21-5754-83A8-FBAB07CF1D7B","softwareVersionExternalIdentifier":"846044774","thirdParty":true,"applicationVariant":"1:MacFamily20,1:15"},
  "parentProc" : "launchd",
  "parentPid" : 1,
  "coalitionName" : "app.ish.iSH",
  "crashReporterKey" : "9778F5AA-205D-100A-1057-C2B4B768E788",
  "wakeTime" : 8053,
  "sleepWakeUUID" : "468F328B-4E9F-4076-8925-007651C0896D",
  "sip" : "enabled",
  "isCorpse" : 1,
  "exception" : {"codes":"0x0000000000000001, 0x00000001b2769514","rawCodes":[1,7289083156],"type":"EXC_BREAKPOINT","signal":"SIGTRAP"},
  "termination" : {"flags":0,"code":5,"namespace":"SIGNAL","indicator":"Trace\/BPT trap: 5","byProc":"exc handler","byPid":77790},
  "asi" : {"AppKit":["Crashing on exception: *** -[__NSDictionaryM setObject:forKey:]: object cannot be nil (key: title)"]},
  "asiBacktraces" : ["0   CoreFoundation                      0x00000001af90eb08 __exceptionPreprocess + 240\n1   libobjc.A.dylib                     0x00000001af659e14 objc_exception_throw + 60\n2   CoreFoundation                      0x00000001af9e0384 -[__NSCFString characterAtIndex:].cold.1 + 0\n3   CoreFoundation                      0x00000001af9ed198 -[__NSDictionaryM setObject:forKey:].cold.3 + 0\n4   CoreFoundation                      0x00000001af847164 -[__NSDictionaryM setObject:forKey:] + 904\n5   UIKitCore                           0x00000001d791c510 -[_UIMenuBarItem properties] + 124\n6   UIKitMacHelper                      0x00000001c5111898 UINSNSMenuItemFromUINSMenuItem + 96\n7   UIKitMacHelper                      0x00000001c511159c _insertUINSMenuItemsIntoNSMenu + 852\n8   UIKitMacHelper                      0x00000001c5110ff4 UINSNSMenuFromUINSMenu + 152\n9   UIKitMacHelper                      0x00000001c5110ec4 -[UINSMenuController _createNSMenu:forContextMenu:] + 92\n10  UIKitMacHelper                      0x00000001c51831d4 -[UINSMenuController _prepareToShowContextMenu:activityItemsConfiguration:] + 144\n11  UIKitMacHelper                      0x00000001c5182b34 -[UINSMenuController showContextMenu:inWindow:atLocationInWindow:activityItemsConfiguration:] + 312\n12  libdispatch.dylib                   0x00000001af5fc604 _dispatch_call_block_and_release + 32\n13  libdispatch.dylib                   0x00000001af5fe1c8 _dispatch_client_callout + 20\n14  libdispatch.dylib                   0x00000001af60c6cc _dispatch_main_queue_drain + 928\n15  libdispatch.dylib                   0x00000001af60c31c _dispatch_main_queue_callback_4CF + 44\n16  CoreFoundation                      0x00000001af8d03e4 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16\n17  CoreFoundation                      0x00000001af88d614 __CFRunLoopRun + 2532\n18  CoreFoundation                      0x00000001af88c490 CFRunLoopRunSpecific + 600\n19  HIToolbox                           0x00000001b84acdd8 RunCurrentEventLoopInMode + 292\n20  HIToolbox                           0x00000001b84acb54 ReceiveNextEventCommon + 564\n21  HIToolbox                           0x00000001b84ac908 _BlockUntilNextEventMatchingListInModeWithFilter + 72\n22  AppKit                              0x00000001b23ebb2c _DPSNextEvent + 860\n23  AppKit                              0x00000001b23ea424 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1328\n24  AppKit                              0x00000001b23dc5f0 -[NSApplication run] + 596\n25  AppKit                              0x00000001b23add08 NSApplicationMain + 1132\n26  AppKit                              0x00000001b2683dd0 +[NSWindow _savedFrameFromString:] + 0\n27  UIKitMacHelper                      0x00000001c510ea7c UINSApplicationMain + 1276\n28  UIKitCore                           0x00000001d77a99f8 UIApplicationMain + 164\n29  iSH                                 0x0000000102c428a8 iSH + 43176\n30  dyld                                0x0000000102fe5088 start + 516"],
  "extMods" : {"caller":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"system":{"thread_create":0,"thread_set_state":0,"task_for_pid":20},"targeted":{"thread_create":0,"thread_set_state":0,"task_for_pid":0},"warnings":0},
  "lastExceptionBacktrace" : [{"imageOffset":1067764,"symbol":"__exceptionPreprocess","symbolLocation":220,"imageIndex":1},{"imageOffset":101908,"symbol":"objc_exception_throw","symbolLocation":60,"imageIndex":2},{"imageOffset":1926020,"symbol":"-[__NSCFString characterAtIndex:].cold.1","symbolLocation":0,"imageIndex":1},{"imageOffset":1978776,"symbol":"-[__NSDictionaryM setObject:forKey:].cold.3","symbolLocation":0,"imageIndex":1},{"imageOffset":250212,"symbol":"-[__NSDictionaryM setObject:forKey:]","symbolLocation":904,"imageIndex":1},{"imageOffset":1533200,"symbol":"-[_UIMenuBarItem properties]","symbolLocation":124,"imageIndex":7},{"imageOffset":30872,"symbol":"UINSNSMenuItemFromUINSMenuItem","symbolLocation":96,"imageIndex":6},{"imageOffset":30108,"symbol":"_insertUINSMenuItemsIntoNSMenu","symbolLocation":852,"imageIndex":6},{"imageOffset":28660,"symbol":"UINSNSMenuFromUINSMenu","symbolLocation":152,"imageIndex":6},{"imageOffset":28356,"symbol":"-[UINSMenuController _createNSMenu:forContextMenu:]","symbolLocation":92,"imageIndex":6},{"imageOffset":496084,"symbol":"-[UINSMenuController _prepareToShowContextMenu:activityItemsConfiguration:]","symbolLocation":144,"imageIndex":6},{"imageOffset":494388,"symbol":"-[UINSMenuController showContextMenu:inWindow:atLocationInWindow:activityItemsConfiguration:]","symbolLocation":312,"imageIndex":6},{"imageOffset":9732,"symbol":"_dispatch_call_block_and_release","symbolLocation":32,"imageIndex":4},{"imageOffset":16840,"symbol":"_dispatch_client_callout","symbolLocation":20,"imageIndex":4},{"imageOffset":75468,"symbol":"_dispatch_main_queue_drain","symbolLocation":928,"imageIndex":4},{"imageOffset":74524,"symbol":"_dispatch_main_queue_callback_4CF","symbolLocation":44,"imageIndex":4},{"imageOffset":812004,"symbol":"__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__","symbolLocation":16,"imageIndex":1},{"imageOffset":538132,"symbol":"__CFRunLoopRun","symbolLocation":2532,"imageIndex":1},{"imageOffset":533648,"symbol":"CFRunLoopRunSpecific","symbolLocation":600,"imageIndex":1},{"imageOffset":212440,"symbol":"RunCurrentEventLoopInMode","symbolLocation":292,"imageIndex":5},{"imageOffset":211796,"symbol":"ReceiveNextEventCommon","symbolLocation":564,"imageIndex":5},{"imageOffset":211208,"symbol":"_BlockUntilNextEventMatchingListInModeWithFilter","symbolLocation":72,"imageIndex":5},{"imageOffset":269100,"symbol":"_DPSNextEvent","symbolLocation":860,"imageIndex":0},{"imageOffset":263204,"symbol":"-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]","symbolLocation":1328,"imageIndex":0},{"imageOffset":206320,"symbol":"-[NSApplication run]","symbolLocation":596,"imageIndex":0},{"imageOffset":15624,"symbol":"NSApplicationMain","symbolLocation":1132,"imageIndex":0},{"imageOffset":2989520,"symbol":"+[NSWindow _savedFrameFromString:]","symbolLocation":0,"imageIndex":0},{"imageOffset":19068,"symbol":"UINSApplicationMain","symbolLocation":1276,"imageIndex":6},{"imageOffset":14840,"symbol":"UIApplicationMain","symbolLocation":164,"imageIndex":7},{"imageOffset":43176,"imageIndex":8},{"imageOffset":20616,"symbol":"start","symbolLocation":516,"imageIndex":9}],
  "faultingThread" : 0,
  "threads" : [{"triggered":true,"id":5312930,"threadState":{"x":[{"value":5529230336},{"value":6125539648},{"value":4},{"value":5529232896},{"value":467},{"value":6125539464},{"value":5},{"value":7241971676,"symbolLocation":0,"symbol":"firstByteMark"},{"value":8682278912,"symbolLocation":8,"symbol":"_OBJC_PROTOCOL_$_NSXPCListenerDelegate"},{"value":14798951100199469288},{"value":1},{"value":5529233363},{"value":4238353408},{"value":2},{"value":65533},{"value":3},{"value":1},{"value":6125540582},{"value":0},{"value":5397581248},{"value":8481438926,"objc-selector":"callStackSymbols"},{"value":5496722704},{"value":8481415168,"objc-selector":"ionDrawerSiriSuggestionsTableViewCellDailyRoutineDelegate"},{"value":8723423232,"symbolLocation":16,"symbol":"OBJC_METACLASS_$__NSTextViewRulerHelper"},{"value":5397531392},{"value":5397143552},{"value":8481420185,"objc-selector":"objectAtIndex:"},{"value":8481426705,"objc-selector":"unsignedIntegerValue"},{"value":7244269580,"symbolLocation":12708,"symbol":"_CFStringCreateByAddingPercentEncodingWithAllowedCharacters.hexchars"}],"flavor":"ARM_THREAD_STATE64","lr":{"value":8796937454427215116},"cpsr":{"value":1610616832},"fp":{"value":6125542208},"sp":{"value":6125542128},"esr":{"value":4060086273,"description":"(Breakpoint) brk 1"},"pc":{"value":7289083156,"matchesCrashFrame":1},"far":{"value":5530481160}},"name":"login-7","queue":"com.apple.main-thread","frames":[{"imageOffset":3929364,"symbol":"-[NSApplication _crashOnException:]","symbolLocation":324,"imageIndex":0},{"imageOffset":3929356,"symbol":"-[NSApplication _crashOnException:]","symbolLocation":316,"imageIndex":0},{"imageOffset":3928620,"symbol":"-[NSApplication reportException:]","symbolLocation":580,"imageIndex":0},{"imageOffset":4626184,"symbol":"uncaughtErrorProc","symbolLocation":156,"imageIndex":0},{"imageOffset":1641388,"symbol":"__handleUncaughtException","symbolLocation":676,"imageIndex":1},{"imageOffset":135972,"symbol":"_objc_terminate()","symbolLocation":132,"imageIndex":2},{"imageOffset":69284,"symbol":"std::__terminate(void (*)())","symbolLocation":20,"imageIndex":3},{"imageOffset":69184,"symbol":"std::terminate()","symbolLocation":64,"imageIndex":3},{"imageOffset":16860,"symbol":"_dispatch_client_callout","symbolLocation":40,"imageIndex":4},{"imageOffset":75468,"symbol":"_dispatch_main_queue_drain","symbolLocation":928,"imageIndex":4},{"imageOffset":74524,"symbol":"_dispatch_main_queue_callback_4CF","symbolLocation":44,"imageIndex":4},{"imageOffset":812004,"symbol":"__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__","symbolLocation":16,"imageIndex":1},{"imageOffset":538132,"symbol":"__CFRunLoopRun","symbolLocation":2532,"imageIndex":1},{"imageOffset":533648,"symbol":"CFRunLoopRunSpecific","symbolLocation":600,"imageIndex":1},{"imageOffset":212440,"symbol":"RunCurrentEventLoopInMode","symbolLocation":292,"imageIndex":5},{"imageOffset":211796,"symbol":"ReceiveNextEventCommon","symbolLocation":564,"imageIndex":5},{"imageOffset":211208,"symbol":"_BlockUntilNextEventMatchingListInModeWithFilter","symbolLocation":72,"imageIndex":5},{"imageOffset":269100,"symbol":"_DPSNextEvent","symbolLocation":860,"imageIndex":0},{"imageOffset":263204,"symbol":"-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:]","symbolLocation":1328,"imageIndex":0},{"imageOffset":206320,"symbol":"-[NSApplication run]","symbolLocation":596,"imageIndex":0},{"imageOffset":15624,"symbol":"NSApplicationMain","symbolLocation":1132,"imageIndex":0},{"imageOffset":2989520,"symbol":"_NSApplicationMainWithInfoDictionary","symbolLocation":24,"imageIndex":0},{"imageOffset":19068,"symbol":"UINSApplicationMain","symbolLocation":1276,"imageIndex":6},{"imageOffset":14840,"symbol":"UIApplicationMain","symbolLocation":164,"imageIndex":7},{"imageOffset":43176,"imageIndex":8},{"imageOffset":20616,"symbol":"start","symbolLocation":516,"imageIndex":9}]},{"id":5313099,"name":"com.apple.uikit.eventfetch-thread","frames":[{"imageOffset":6352,"symbol":"mach_msg_trap","symbolLocation":8,"imageIndex":10},{"imageOffset":7488,"symbol":"mach_msg","symbolLocation":76,"imageIndex":10},{"imageOffset":543772,"symbol":"__CFRunLoopServiceMachPort","symbolLocation":372,"imageIndex":1},{"imageOffset":536780,"symbol":"__CFRunLoopRun","symbolLocation":1180,"imageIndex":1},{"imageOffset":533648,"symbol":"CFRunLoopRunSpecific","symbolLocation":600,"imageIndex":1},{"imageOffset":410128,"symbol":"-[NSRunLoop(NSRunLoop) runMode:beforeDate:]","symbolLocation":236,"imageIndex":11},{"imageOffset":979548,"symbol":"-[NSRunLoop(NSRunLoop) runUntilDate:]","symbolLocation":124,"imageIndex":11},{"imageOffset":20852,"symbol":"-[UIEventFetcher threadMain]","symbolLocation":156,"imageIndex":7},{"imageOffset":378652,"symbol":"__NSThread__start__","symbolLocation":808,"imageIndex":11},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313118,"name":"init-1","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":177856,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313124,"name":"getty-5","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":244796,"imageIndex":8},{"imageOffset":188536,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313125,"name":"getty-6","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":244796,"imageIndex":8},{"imageOffset":188536,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313142,"name":"com.apple.NSEventThread","frames":[{"imageOffset":6352,"symbol":"mach_msg_trap","symbolLocation":8,"imageIndex":10},{"imageOffset":7488,"symbol":"mach_msg","symbolLocation":76,"imageIndex":10},{"imageOffset":543772,"symbol":"__CFRunLoopServiceMachPort","symbolLocation":372,"imageIndex":1},{"imageOffset":536780,"symbol":"__CFRunLoopRun","symbolLocation":1180,"imageIndex":1},{"imageOffset":533648,"symbol":"CFRunLoopRunSpecific","symbolLocation":600,"imageIndex":1},{"imageOffset":1763060,"symbol":"_NSEventThread","symbolLocation":196,"imageIndex":0},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313146,"name":"login-7","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":158200,"imageIndex":8},{"imageOffset":159140,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313148,"name":"ash-8","frames":[{"imageOffset":31808,"symbol":"kevent","symbolLocation":8,"imageIndex":10},{"imageOffset":257876,"imageIndex":8},{"imageOffset":260652,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313162,"name":"getty-11","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":244796,"imageIndex":8},{"imageOffset":188536,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313215,"name":"getty-12","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":244796,"imageIndex":8},{"imageOffset":188536,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313224,"name":"getty-13","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":244796,"imageIndex":8},{"imageOffset":188536,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5313228,"name":"getty-14","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":263696,"imageIndex":8},{"imageOffset":263412,"imageIndex":8},{"imageOffset":244796,"imageIndex":8},{"imageOffset":188536,"imageIndex":8},{"imageOffset":142384,"imageIndex":8},{"imageOffset":146088,"imageIndex":8},{"imageOffset":146320,"imageIndex":8},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5318963,"frames":[{"imageOffset":8312,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":12}]},{"id":5319700,"frames":[{"imageOffset":8312,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":12}]},{"id":5320515,"name":"JavaScriptCore libpas scavenger","frames":[{"imageOffset":21136,"symbol":"__psynch_cvwait","symbolLocation":8,"imageIndex":10},{"imageOffset":30780,"symbol":"_pthread_cond_wait","symbolLocation":1236,"imageIndex":12},{"imageOffset":19206988,"symbol":"scavenger_thread_main","symbolLocation":1220,"imageIndex":13},{"imageOffset":29292,"symbol":"_pthread_start","symbolLocation":148,"imageIndex":12},{"imageOffset":8332,"symbol":"thread_start","symbolLocation":8,"imageIndex":12}]},{"id":5320645,"frames":[{"imageOffset":8312,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":12}]},{"id":5320646,"frames":[{"imageOffset":8312,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":12}]},{"id":5320674,"frames":[{"imageOffset":8312,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":12}]},{"id":5320675,"frames":[{"imageOffset":8312,"symbol":"start_wqthread","symbolLocation":0,"imageIndex":12}]}],
  "usedImages" : [
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7285153792,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.AppKit",
    "size" : 15433728,
    "uuid" : "153590c1-624c-38fb-8ffb-ba751bccbd16",
    "path" : "\/System\/Library\/Frameworks\/AppKit.framework\/Versions\/C\/AppKit",
    "name" : "AppKit",
    "CFBundleVersion" : "2113.40.127"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7239409664,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.CoreFoundation",
    "size" : 5529600,
    "uuid" : "8e1809ff-f5b1-3d9e-be15-f2e7b1d8ce51",
    "path" : "\/System\/Library\/Frameworks\/CoreFoundation.framework\/Versions\/A\/CoreFoundation",
    "name" : "CoreFoundation",
    "CFBundleVersion" : "1858.112"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7237537792,
    "size" : 253952,
    "uuid" : "4b5ab8fb-60e0-3992-ba0e-a073a3fa86a7",
    "path" : "\/usr\/lib\/libobjc.A.dylib",
    "name" : "libobjc.A.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7238758400,
    "size" : 98304,
    "uuid" : "e0fc64a3-790d-34b3-a092-432a0e6b1c82",
    "path" : "\/usr\/lib\/libc++abi.dylib",
    "name" : "libc++abi.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7237246976,
    "size" : 290816,
    "uuid" : "dc048e3b-e023-3d17-afe5-4ff3dc625608",
    "path" : "\/usr\/lib\/system\/libdispatch.dylib",
    "name" : "libdispatch.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7386664960,
    "CFBundleShortVersionString" : "2.1.1",
    "CFBundleIdentifier" : "com.apple.HIToolbox",
    "size" : 3358720,
    "uuid" : "4a1d1f30-c79d-3502-b51f-9792adbd8e06",
    "path" : "\/System\/Library\/Frameworks\/Carbon.framework\/Versions\/A\/Frameworks\/HIToolbox.framework\/Versions\/A\/HIToolbox",
    "name" : "HIToolbox"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7601168384,
    "CFBundleShortVersionString" : "1.0",
    "CFBundleIdentifier" : "com.apple.UIKitMacHelper",
    "size" : 1073152,
    "uuid" : "cf1d8545-c12b-3515-9519-b9840e38052c",
    "path" : "\/System\/Library\/PrivateFrameworks\/UIKitMacHelper.framework\/Versions\/A\/UIKitMacHelper",
    "name" : "UIKitMacHelper",
    "CFBundleVersion" : "5522.2.400"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7910088704,
    "CFBundleShortVersionString" : "1.0",
    "CFBundleIdentifier" : "com.apple.UIKitCore",
    "size" : 24162304,
    "uuid" : "aa248023-fcd7-308f-bcdc-b93116ba5a2b",
    "path" : "\/System\/iOSSupport\/System\/Library\/PrivateFrameworks\/UIKitCore.framework\/Versions\/A\/UIKitCore",
    "name" : "UIKitCore",
    "CFBundleVersion" : "5522.2.400"
  },
  {
    "source" : "P",
    "arch" : "arm64",
    "base" : 4341334016,
    "CFBundleShortVersionString" : "1.2.3",
    "CFBundleIdentifier" : "app.ish.iSH",
    "size" : 671744,
    "uuid" : "5f72cbc8-f1ca-34a3-8882-bf0a879afe83",
    "path" : "\/private\/var\/folders\/*\/iSH.app\/iSH",
    "name" : "iSH",
    "CFBundleVersion" : "298"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 4345167872,
    "size" : 393216,
    "uuid" : "fbb89662-e6f2-3434-b542-f75185ac5e74",
    "path" : "\/usr\/lib\/dyld",
    "name" : "dyld"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7238856704,
    "size" : 229376,
    "uuid" : "1d7b3b8e-75a1-34ea-aa52-9f7c23155c55",
    "path" : "\/usr\/lib\/system\/libsystem_kernel.dylib",
    "name" : "libsystem_kernel.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7255117824,
    "CFBundleShortVersionString" : "6.9",
    "CFBundleIdentifier" : "com.apple.Foundation",
    "size" : 4132864,
    "uuid" : "886a3487-88a6-30bb-b179-1a92996c67cf",
    "path" : "\/System\/Library\/Frameworks\/Foundation.framework\/Versions\/C\/Foundation",
    "name" : "Foundation",
    "CFBundleVersion" : "1858.112"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 7239086080,
    "size" : 53248,
    "uuid" : "cee8bc77-6923-34d9-89a3-6f8f7279605e",
    "path" : "\/usr\/lib\/system\/libsystem_pthread.dylib",
    "name" : "libsystem_pthread.dylib"
  },
  {
    "source" : "P",
    "arch" : "arm64e",
    "base" : 8172883968,
    "CFBundleShortVersionString" : "17613",
    "CFBundleIdentifier" : "com.apple.JavaScriptCore",
    "size" : 20430848,
    "uuid" : "f13513f2-9b55-396d-b981-add74643fb96",
    "path" : "\/System\/iOSSupport\/System\/Library\/Frameworks\/JavaScriptCore.framework\/Versions\/A\/JavaScriptCore",
    "name" : "JavaScriptCore",
    "CFBundleVersion" : "17613.1.17.1.13"
  }
],
  "sharedCache" : {
  "base" : 7235796992,
  "size" : 3136438272,
  "uuid" : "1df3dfc1-141a-35d0-a4e5-f1e113894c6e"
},
  "vmSummary" : "ReadOnly portion of Libraries: Total=1.1G resident=0K(0%) swapped_out_or_unallocated=1.1G(100%)\nWritable regions: Total=575.4M written=0K(0%) resident=0K(0%) swapped_out=0K(0%) unallocated=575.4M(100%)\n\n                                VIRTUAL   REGION \nREGION TYPE                        SIZE    COUNT (non-coalesced) \n===========                     =======  ======= \nAccelerate framework               256K        2 \nActivity Tracing                   256K        1 \nCG backing stores                 3840K        4 \nCG image                            96K        4 \nCG raster data                     832K        5 \nColorSync                          640K       27 \nCoreAnimation                     37.5M       37 \nCoreGraphics                        32K        2 \nCoreUI image data                 1360K        9 \nDispatch continuations            80.0M        1 \nFoundation                          48K        2 \nImage IO                          70.0M       15 \nKernel Alloc Once                   32K        1 \nMALLOC                           155.3M       68 \nMALLOC guard page                  192K        9 \nMALLOC_MEDIUM (reserved)         104.0M        1         reserved VM address space (unallocated)\nSQLite page cache                  320K        5 \nSTACK GUARD                       56.3M       19 \nStack                             17.5M       19 \nVM_ALLOCATE                       1296K       54 \nVM_ALLOCATE (reserved)              16K        1         reserved VM address space (unallocated)\nWebKit Malloc                    192.0M        4 \n__AUTH                            3556K      438 \n__AUTH_CONST                      23.8M      650 \n__CTF                               759        1 \n__DATA                            20.6M      662 \n__DATA_CONST                      24.8M      677 \n__DATA_DIRTY                      2626K      293 \n__FONT_DATA                          4K        1 \n__LINKEDIT                       585.4M       26 \n__OBJC_CONST                      5633K      406 \n__OBJC_RO                         82.9M        1 \n__OBJC_RW                         3152K        1 \n__TEXT                           572.0M      697 \n__UNICODE                          592K        1 \ndyld private memory               1024K        1 \nmapped file                      205.2M       93 \nshared memory                     1360K       14 \n===========                     =======  ======= \nTOTAL                              2.2G     4252 \nTOTAL, minus reserved VM space     2.1G     4252 \n",
  "legacyInfo" : {
  "threadTriggered" : {
    "name" : "login-7",
    "queue" : "com.apple.main-thread"
  }
},
  "trialInfo" : {
  "rollouts" : [
    {
      "rolloutId" : "60356660bbe37970735c5624",
      "factorPackIds" : {

      },
      "deploymentId" : 240000027
    },
    {
      "rolloutId" : "607844aa04477260f58a8077",
      "factorPackIds" : {
        "SIRI_MORPHUN_ASSETS" : "6103050cbfe6dc472e1c982a"
      },
      "deploymentId" : 240000066
    }
  ],
  "experiments" : [

  ]
}
}

Model: MacBookPro18,4, BootROM 7459.101.3, proc 10:8:2 processors, 64 GB, SMC 
Graphics: Apple M1 Max, Apple M1 Max, Built-In
Display: PHL 279M1RV, 5120 x 2880 (5K/UHD+ - Ultra High Definition Plus), Main, MirrorOff, Online
Memory Module: LPDDR5
AirPort: Wi-Fi, wl0: Feb  8 2022 01:51:44 version 20.90.39.0.8.7.114 FWID 01-510a2aee
Bluetooth: Version (null), 0 services, 0 devices, 0 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
USB Device: USB31Bus
USB Device: USB31Bus
USB Device: USB2.0 Hub
USB Device: USB Receiver
USB Device: USB2.0 Hub
USB Device: USB 2.0 BILLBOARD
USB Device: Lighting Board
USB Device: HP 4K Webcam
USB Device: USB Billboard Device
USB Device: USB31Bus
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.
Thunderbolt Bus: MacBook Pro, Apple Inc.

```
